youtube_transcript_api/_cli.py

```diff
@@ -26,19 +25,24 @@ def run(self):
         transcripts = []
         exceptions = []
 
+        Formatter = formats.get_formatter(parsed_args.format)
```

Comment:
In regards to the `parsed_args.format` being `None` this would fail if I had not chosen to use an underlying `defaultdict` with `JSONTranscriptFormatter` as the default for the TranscriptFormatterFactory `._formatters` attribute. I kind of hid this away but because it is a defaultdict using bracket notation such as this line:

```python
formats = TranscriptFormatterFactory()
formats.get_formatter(None) # <-- should default return to JSONTranscriptFormatter
```
So it shouldn't cause an error. I kept fighting myself on whether I wanted to just pass `'json'` as the default or just use `None` because either way I was still going to have to specify a default action to take for a Formatter or raise an Exception. But the initial discussion was to default to JSON so that was my thinking there.

Reply:
Ah yeah, didn't notice that.
However, the pprinted dict is the default in the current version, not JSON. I'd like to keep it that way, to not break the API more than necessary. Therefore, you'd have to also implement a `PrettyPrintFormatter`, as already pointed out in the other comment, and then make it the default. Passing around `'json'` won't be an issue, if we don't make the formatters part of `get_transcript`.